### PR TITLE
#1057 #1068 Add service discovery logs

### LINKED ISSUE #1057: Kubernetes and Consul Integration question
Hello,

I am using Ocelot as an API gateway in AKS with Consul deployed inside the cluster for service discovery. I am using connect sidecar injection with ACL tokens for my pods. So my question is how to best integrate Ocelot with Consul in this scenario. This [issue](https://github.com/ThreeMammals/Ocelot/issues/332) references what I am trying to do but doesn't lay out a preferred route. I see three potential solutions.
1) Annotate the ocelot pod for connect injection and provide the list of required services to the connect-service-upstreams. This would create localhost bindings for each service and then I would use Ocelots normal configuration (without consul integration)
```json
{
    "DownstreamPathTemplate": "/api/posts/{postId}",
    "DownstreamScheme": "https",
    "DownstreamHostAndPorts": [
            {
                "Host": "127.0.0.1",
                "Port": {servicePortConfiguredInUpstreams},
            }
        ],
    "UpstreamPathTemplate": "/posts/{postId}",
    "UpstreamHttpMethod": [ "Put", "Delete" ]
}
```

2) Don't use connect inject on the Ocelot pod, use the Consul integration as defined in the configuration
```json
"ServiceDiscoveryProvider": {
 "Host": "consul",
 "Port": 8500,
 "Token": "{aclToken}",
 "Namespace": "{consulns}"
}
```

3) Use connect inject and the consul integration. In this scenario connect-inject is setup, but the service being injected is the consul service.
```json
{
      "DownstreamPathTemplate": "/api/{everything}",
      "DownstreamScheme": "http",
      "ServiceName": "{requestedService}",
      "UpstreamPathTemplate": "/api/c/{everything}",
      "UpstreamHttpMethod": [],
      "AuthenticationOptions": {
        "AuthenticationProviderKey": "IdentityApiKey",
        "AllowedScopes": []
      }
```

```json
"ServiceDiscoveryProvider": {
      "Host": "127.0.0.1",
      "Port": 8500,
      "Token": "{aclToken}",
      "Type": "Consul"
    }
```
Of the three I think I prefer the first option as it allows consul to manage it's ACL tokens, but is there a recommended best practice for this scenario?

Thanks!

### PR DESCRIPTION
**Pull request by jlukawska**
## Closes #1057 #1068 
- #1057 
- #1068 

## Proposed Changes

The issues regarding communication with Consul are quite common. Currently the Ocelot logs don't show the basic information about this communication. I think it could be very useful for debugging purposes.

### CODE REVIEW

**Review by EngRajabi — APPROVED** [review]
ok

**Comment by raman-m** [comment]
Thank you for the PR!

Could you [Sync fork](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) please? So, your **develop** branch is outdated!

**Comment by raman-m** [comment]
The [feature branch](/jlukawska/Ocelot/tree/add-service-discovery-logs) has been [rebased](#event-10062627483) onto [ThreeMammals:develop](../tree/develop)!
Welcome to code review!

**Review by wast — DISMISSED** [review]

**Review by RaynaldM — APPROVED** [review]

File: src/Ocelot.Provider.Consul/Consul.cs [hunk 1]
```diff
@@ -49,6 +51,8 @@ public async Task<List<Service>> Get()
                         var serviceNode = nodes.Response.FirstOrDefault(n => n.Address == serviceEntry.Service.Address);
                         services.Add(BuildService(serviceEntry, serviceNode));
                     }
+
+                    _logger.LogDebug($"Consul answer: Address: {serviceEntry.Service.Address}, Port: {serviceEntry.Service.Port}");
```
Comment (raman-m) [inline src/Ocelot.Provider.Consul/Consul.cs]: Long expressions of property reading with multiple references.
```suggestion
                    var address = serviceEntry.Service.Address;
                    var port = serviceEntry.Service.Port;
                    _logger.LogDebug($"Consul answer: Address: {address}, Port: {port}");
```

**Review by raman-m — DISMISSED** [review]

**Comment by raman-m** [comment]
@TomPallister 
This PR is ready to merge.

**Review by RaynaldM — DISMISSED** [review]

File: src/Ocelot.Provider.Consul/Consul.cs [hunk 2]
```diff
@@ -31,6 +31,8 @@ public Consul(ConsulRegistryConfiguration config, IOcelotLoggerFactory factory,
 
         public async Task<List<Service>> Get()
         {
+            _logger.LogDebug($"Querying Consul {(_consul as ConsulClient)?.Config.Address} about a service: {_config.KeyOfServiceInConsul}");
```
[the remark this answers is not in the document]
Comment (raman-m) [reply src/Ocelot.Provider.Consul/Consul.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1366/commits/5a5f1c01949e74494aa3e03bb5a9d99fce4b95ce

File: src/Ocelot.Provider.Consul/Consul.cs (hunk 1 above)
Comment (raman-m) [reply src/Ocelot.Provider.Consul/Consul.cs]: ✔️ Fixed in https://github.com/ThreeMammals/Ocelot/pull/1366/commits/5a5f1c01949e74494aa3e03bb5a9d99fce4b95ce

**Review by raman-m — APPROVED** [review]

**Review by RaynaldM — APPROVED** [review]

**Pull request merged**